elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/mapping/CaseMappingGenerator.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ccd.definition.store.elastic.mapping;
+
+import com.google.gson.stream.JsonWriter;
+import lombok.extern.slf4j.Slf4j;
+import org.jooq.lambda.Unchecked;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.elastic.mapping.type.TypeMappingGenerator;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+
+@Component
+@Slf4j
+public class CaseMappingGenerator extends MappingGenerator {
+
+    public String generateMapping(CaseTypeEntity caseType) {
+        log.debug("creating mapping for case type: {}", caseType.getReference());
+
+        String mapping = newJson(Unchecked.consumer((JsonWriter jw) -> {
+            jw.name("dynamic");
+            jw.value(config.getDynamic());
+            jw.name(PROPERTIES);
+            jw.beginObject();
+            propertiesMapping(jw);
+            dataMapping(jw, caseType);
+            dataClassificationMapping(jw, caseType);
+            jw.endObject();
+        }));
+
+        log.debug("generated mapping for case type {}: {}", caseType.getReference(), mapping);
+        return mapping;
+    }
+
+    private void propertiesMapping(JsonWriter jw) {
+        log.debug("generating case properties mapping");
+        config.getCasePredefinedMappings().forEach(Unchecked.biConsumer((property, mapping) -> {
+            jw.name(property);
+            jw.jsonValue(mapping);
+            log.debug("property: {}, mapping: {}", property, mapping);
+        }));
+    }
+
+    private void dataMapping(JsonWriter jw, CaseTypeEntity caseType) throws IOException {
+        log.debug("generating case data mapping");
+        jw.name(DATA);
+        genericDataMapping(jw, caseType, typeMapper -> typeMapper::dataMapping);
+    }
+
+    private void dataClassificationMapping(JsonWriter jw, CaseTypeEntity caseType) throws IOException {
+        log.debug("generating case data classification mapping");
+        jw.name(DATA_CLASSIFICATION);
+        genericDataMapping(jw, caseType, typeMapper -> typeMapper::dataClassificationMapping);
+    }
+
+    private void genericDataMapping(JsonWriter jw, CaseTypeEntity caseType,
+                                    Function<TypeMappingGenerator, Function<CaseFieldEntity, String>> mappingMethodSelection) throws IOException {
+        jw.beginObject();
+        jw.name(PROPERTIES);
+        jw.beginObject();
+        List<CaseFieldEntity> fields = caseType.getCaseFields().stream().filter(field -> !shouldIgnore(field)).collect(toList());
```

Comment:
the thing is that this concept of ignoring fields is only used in here. If this could have been a general utility method I would have added it directly
in CaseType. But really I don't see other uses cases of ignoring a case fields